### PR TITLE
refactor proving

### PR DESCRIPTION
**Pull request by AndrewCLu**
- [x] Make all proving sequential for mobile browser memory issues
- [x] Add progress counter hooks to all proving functions
- [x] Remove custom team leaderboard proving code

### CODE REVIEW

**Review by vb7401 — COMMENTED** [review]
LGTM after refactors

File: apps/jubmoji-quest/src/components/powers/RedirectPower.tsx [hunk 1]
```diff
@@ -4,17 +4,29 @@ import toast from "react-hot-toast";
...
+
   const onRedirect = async () => {
     try {
       const serializedProof = await powerMutation.mutateAsync({
         power,
         jubmojis,
+        onUpdateProvingState,
       });
 
       const url = new URL(redirectUrl!);
```
Comment (vb7401) [inline apps/jubmoji-quest/src/components/powers/RedirectPower.tsx:32]: Redirect URL needs to start with "https://" to work

**Review by vb7401 — APPROVED** [review]
LGTM after all comments are responded to. All proof types and powers have been tested!

File: apps/jubmoji-quest/src/components/powers/RedirectPower.tsx (hunk 1 above)
Comment (AndrewCLu) [reply apps/jubmoji-quest/src/components/powers/RedirectPower.tsx:32]: documented

File: packages/jubmoji-api/src/proofs.ts [hunk 2]
```diff
@@ -387,35 +427,40 @@ export class TeamLeaderboard
...
+    const collectionMembershipProofs = [];
+    for (let i = 0; i < collectionJubmojis.length; i++) {
+      const jubmoji = collectionJubmojis[i];
       const { sig, msgHash, pubKey, R, T, U } =
         getMembershipProofArgsFromJubmoji(jubmoji);
-      sigs.push(sig);
-      msgHashes.push(msgHash);
-      publicInputs.push({ R, T, U });
-      indices.push(this.collectionPubKeys.indexOf(pubKey));
+      const collectionMerkleProof = await getMerkleProofFromCache(
```
[the remark this answers is not in the document]
Comment (AndrewCLu) [reply packages/jubmoji-api/src/proofs.ts:440]: yes when caching is implemented

File: apps/jubmoji-quest/src/components/powers/RedirectPower.tsx [hunk 3]
```diff
@@ -4,17 +4,29 @@ import toast from "react-hot-toast";
...
+import { useState } from "react";
+import { ProvingState } from "jubmoji-api";
 
 const RedirectPower = ({ power, jubmojis }: PowerContentProps) => {
+  const [provingState, setProvingState] = useState<ProvingState>();
   const powerParams = power.powerParams as Prisma.JsonObject;
   const redirectUrl = powerParams.redirectUrl as string | undefined;
   const powerMutation = useRedirectPowerMutation();
 
+  const onUpdateProvingState = (newProvingState: ProvingState) => {
```
[the remark this answers is not in the document]
Comment (AndrewCLu) [reply apps/jubmoji-quest/src/components/powers/RedirectPower.tsx:16]: added

**Pull request merged**